spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesExternalNameServiceInstance.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.discovery;
+
+import java.net.URI;
+import java.util.Map;
+
+import org.springframework.cloud.client.ServiceInstance;
+
+/**
+ * Type of {@link org.springframework.cloud.client.ServiceInstance} when
+ * "spec.type=ServiceInstance".
+ *
+ * @author wind57
+ */
+public record KubernetesExternalNameServiceInstance(String serviceId, String host, String instanceId,
```

Comment:
added a new type of `ServiceInstance`

Reply:
Why a new one?

Reply:
excellent question! as usual. How are users supposed to distinguish between an ExternalName ServiceInstance and one that is based on endpoints? I had the idea to have a `type` here that will present that information...

Reply:
What about via the type in the metadata?

Reply:
I feel like an idiot, you're right. Will change, thx a lot!

Reply:
Not an idiot at all, this is why we review code!